lib/matplotlib/axes/_axes.py

```diff
@@ -7234,6 +7234,11 @@ def specgram(self, x, NFFT=None, Fs=None, Fc=None, detrend=None,
         Z = np.flipud(Z)
 
         if xextent is None:
+            # define default values:
```

Comment:
Can you move this up to the top?  I think it is better to have this over-ride the defaults if we change the defaults in `mlab.specgram` than to have this silently get out of sync (it is unlikely we will make that change, but this just smell very brittle to me).